### ISSUE TITLE
Use OVN provider LB when available in front of ingress

### ISSUE THREAD

**Issue by garloff** [SovereignCloudStack/k8s-cluster-api-provider#325]
As an ingress user, I want to see real client IPs in the ingress.
To make this work without proxy-protocol workarounds, we must ensure 
- [ ] that we use externalTrafficPolicy: local (otherwise we see the kube-proxy forwarded source IPs)
- [ ] have a health-monitor enabled (otherwise the nodes without the nginx will receive and drop traffic)
- [ ] have a loadbalancer that does preserve the client IPs (by working on layer 3 or similar ways to steer the flows)

The octavia load balancer implemented by the OVN provider does preserve the client IPs nicely.

To use it, we can set lb-provider=ovn in [loadbalancer] sertion of cloud.conf, of course only after testing that it is supported by the underlying infra.

See also issues/#268 for figuring out how to make health monitors work.

**Comment by garloff** [comment]
OVN LB is not ready yet:
* It only support TCP Health-Mon while OCCM unconditionally wants an HTTP health-mon
* The health-mon is ineffective when the service is access via FIP (it work for VIPs)
See [issues/#268](https://github.com/SovereignCloudStack/issues/issues/268)

**Comment by garloff** [comment]
PR #351 added the code to use the OVN provider LB in front of ingress (or other services).
As it does not work yet, it's currently guarded by an `--allow-preview-features` switch.

**Comment by garloff** [comment]
Sidenote: Testing zed release (on a cloud from OSISM):
* Manually creating OVN LB with VIP in the same network as the members, healthmon for backend members, attaching FIP, the access via the FIP still shows the same issue not correctly taking out the broken members of the LB rotation, only the VIP accesses.
* On the other hand, using the openstack-health-monitor with -LP ovn, the accesses via FIP *are* taken out when the http servers on the members are killed. The VIP here lives in a different net than the members (and they are connected via a router).
Whether it's really the network location of the VIP that makes the difference is unknown, there may be other differences.

**Comment by garloff** [comment]
The health-mon not working for the floating IPs *should* be fixed in zed since early March, see 
commit https://review.opendev.org/c/openstack/ovn-octavia-provider/+/873860
(commit https://review.opendev.org/c/openstack/ovn-octavia-provider/+/875595 against Zed)
which fixes https://bugs.launchpad.net/neutron/+bug/1997418/
which appears to be a duplicate of https://bugs.launchpad.net/neutron/+bug/1956035.

Assuming that the fix is effective and complete, this leaves the inability to create HTTP health mon (which OCCM unconditionally requests) as the final challenge to have a great LB in front of ingress.
